### PR TITLE
Rectangle Drawing Tool

### PR DESCRIPTION
**Pull request by zbennett10**
Hey guys - this one is not ready yet, but I wanted to get some eyes on it from the maintainers as I'm encountering a bug that is stumping me. The rectangle drawing tool is mostly complete. If you pull up the demo app, you will see a new tool available on the toolbar for drawing rectangles. This works - however, when dragging rectangles that have already been drawn I encounter something weird with positioning where the rectangle either gets shorter and wider or taller and skinnier depending on which way you drag (left or right). @eitanfr Would you mind having a look at the `editable-rectangle` model that I created. I took the `moveShape` functionality from the `editable-hippodrome` model.

Thanks! Once we get this bug fixed, I can update the readme, changelog, etc.

This is in reference to #254

### CODE REVIEW

**Comment by zbennett10** [comment]
Actually - just figured this bug out -- update to follow shortly.

File: projects/angular-cesium/src/lib/angular-cesium-widgets/models/editable-rectangle.ts [hunk 1]
```diff
@@ -141,10 +141,13 @@ export class EditableRectangle extends AcEntity {
       this.lastDraggedToPosition = startMovingPosition;
     }
 
-    const delta = GeoUtilsService.getPositionsDelta(this.lastDraggedToPosition, draggedToPosition);
+    const lastDraggedCartographic = Cesium.Cartographic.fromCartesian(this.lastDraggedToPosition);
+    const draggedToPositionCartographic = Cesium.Cartographic.fromCartesian(draggedToPosition);
     this.getRealPoints().forEach(point => {
-      const newPos = GeoUtilsService.addDeltaToPosition(point.getPosition(), delta);
-      point.setPosition(newPos);
+      const cartographic = Cesium.Cartographic.fromCartesian(point.getPosition());
```
Comment (zbennett10) [inline projects/angular-cesium/src/lib/angular-cesium-widgets/models/editable-rectangle.ts]: Converting the drag positions to cartographics and then altering the points based on the delta calculated from them means that, in 2d mode, the rectangle's width doesn't change now and it looks good. In 3d mode, dragging left to right and vice-versa doesn't alter the height of the rectangle now. However, when you drag the rectangle near the poles, the rectangle begins to wrap around them.

Comment (zbennett10) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/models/editable-rectangle.ts]: I *think* this is due to Rectangles being defined with radians.

Comment (zbennett10) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/models/editable-rectangle.ts]: Yeah, it looks like this is expected behavior with RectangleGraphics -> 

> Describes graphics for a Rectangle. The rectangle conforms to the curvature of the globe and can be placed on the surface or at altitude and can optionally be extruded into a volume.

File: projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+    outlineColor: Cesium.Color.BLACK.withAlpha(0.2),
+    outlineWidth: 1,
+    pixelSize: 13,
+    virtualPointPixelSize: 8,
+    show: true,
+    showVirtual: true,
+    disableDepthTestDistance: Number.POSITIVE_INFINITY,
+  },
+  rectangleProps: {
+    material: Cesium.Color.CORNFLOWERBLUE.withAlpha(0.4),
```
Comment (eitanfr) [inline projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts]: Played with it and it looks good!
can you please support for rectangle outline props, and to add white outline as a default option.

Comment (zbennett10) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts]: Will do

Comment (zbennett10) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts]: @eitanfr Do you want me to add in an icon for the rectangle drawer in the demo app or will you guys take care of that?

Comment (eitanfr) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts]: yes, i will be happy if could do that. 
somthing like https://www.iconfinder.com/icons/755913/design_draw_graphic_rectangle_shape_tool_icon

Comment (zbennett10) [reply projects/angular-cesium/src/lib/angular-cesium-widgets/services/entity-editors/rectangles-editor/rectangles-editor.service.ts]: Ok - I'll leave the icon up to you then!

**Comment by zbennett10** [comment]
Ok @eitanfr rectangleProps have been augmented and I added a rectangle icon. Let me know if you need anything else!

**Comment by zbennett10** [comment]
@eitanfr Hey just pinging you on this one! Do you need anything else?

**Pull request merged**